core/src/main/java/feast/core/controller/CoreServiceRestController.java

```diff
@@ -0,0 +1,159 @@
+  private SpecService specService;
+  private JobService jobService;
+  private StatsService statsService;
+  private AccessManagementService accessManagementService;
+
+  @Autowired
+  public CoreServiceRestController(
+      FeastProperties feastProperties,
+      SpecService specService,
+      JobService jobService,
+      StatsService statsService,
+      AccessManagementService accessManagementService) {
+    this.feastProperties = feastProperties;
+    this.specService = specService;
+    this.jobService = jobService;
+    this.statsService = statsService;
+    this.accessManagementService = accessManagementService;
+    this.jsonPrinter = JsonFormat.printer();
+  }
+
+  @RequestMapping(value = "/version", method = RequestMethod.GET)
+  @ResponseBody
+  public String getVersion() throws InvalidProtocolBufferException {
+    GetFeastCoreVersionResponse response =
+        GetFeastCoreVersionResponse.newBuilder().setVersion(feastProperties.getVersion()).build();
+    return jsonPrinter.print(response);
+  }
+
+  @RequestMapping(
+      value = "/project/{project}/feature-set/{featureSetName}",
+      method = RequestMethod.GET)
+  @ResponseBody
+  public String getFeatureSet(@PathVariable String project, @PathVariable String featureSetName)
+      throws InvalidProtocolBufferException {
+    GetFeatureSetRequest request =
+        GetFeatureSetRequest.newBuilder().setProject(project).setName(featureSetName).build();
+    return jsonPrinter.print(specService.getFeatureSet(request));
+  }
+
+  @RequestMapping(value = "/feature-sets", method = RequestMethod.GET)
+  @ResponseBody
+  public String listFeatureSets(
+      @RequestParam String project, @RequestParam(defaultValue = "*") String name)
+      throws InvalidProtocolBufferException {
+    ListFeatureSetsRequest.Filter filter =
+        ListFeatureSetsRequest.Filter.newBuilder()
+            .setProject(project)
+            .setFeatureSetName(name)
+            .build();
+    return jsonPrinter.print(specService.listFeatureSets(filter));
+  }
+
+  @RequestMapping(value = "/features", method = RequestMethod.GET)
+  @ResponseBody
+  public String listFeatures(
+      @RequestParam String[] entities, @RequestParam(defaultValue = "default") String project)
+      throws InvalidProtocolBufferException {
+    ListFeaturesRequest.Filter filter =
+        ListFeaturesRequest.Filter.newBuilder()
+            .setProject(project)
+            .addAllEntities(Arrays.asList(entities))
+            .build();
+    return jsonPrinter.print(specService.listFeatures(filter));
+  }
+
+  @RequestMapping(value = "/feature-statistics", method = RequestMethod.GET)
+  @ResponseBody
+  public String getFeatureStatistics(
+      @RequestParam String feature_set_id,
+      @RequestParam(required = false) Optional<String[]> features,
+      @RequestParam(required = false) Optional<String> store,
+      @RequestParam(name = "start_date", required = false) Optional<Timestamp> startDate,
```

Comment:
Is there a reason you are mixing camel and snake case?

Reply:
An oversight here. Changed to camel case.

Reply:
Currently the query params are in snake case while as parameters for the controller methods, camel case is used.